### DISCUSSION TITLE
Middleware roles question

### DISCUSSION THREAD

**Discussion by oriceon** [spatie/laravel-permission#2000] [Q&A]
If i have

```
middleware('role:1|2|3')
```

Code will not work as espected because of Spatie\Permission\Traits\HasRoles

```
if (is_int($roles)) {
```

So, in RoleMiddleware, $roles should be rewriten in order to get out with ids as int not as string.

**Comment by erikn69** [comment]
https://spatie.be/docs/laravel-permission/v5/basic-usage/middleware
```php
Route::group(['middleware' => ['role:super-admin|writer']], function () {
    //
});
```
Only works for names, the reason is for let it be roles with numbers in name, example: `Role(id=1,name='5',guard_name='web')`

**Reply by oriceon** [reply]
Ok. I get it.

I made a custom RoleMiddleware with implementation i need and is works ok.
